app/user_views.py

```diff
@@ -1,42 +1,60 @@
-from flask_login import logout_user, current_user, login_required
-from flask import render_template, url_for, redirect, flash, request
+from flask_login import current_user, login_required
+from flask import render_template, url_for, redirect, flash, request, abort
 from flask import current_app as app
 from sqlalchemy import insert
-from .models import db, Rehearsal
-from .forms import RehearsalForm
+from .models import db, Rehearsal, User
+from .forms import RehearsalForm, OrderForm
 import datetime as dt
 
 
-@app.route('/logout')
-def logout():
-    logout_user()
-    return redirect(url_for('index'))
-
-
-@app.route("/all-rehearsals")
+@app.route("/all-rehearsals", methods=["GET", "POST"])
 @login_required
 def get_all_rehearsals():
-    rehearsals = Rehearsal.query.all()
+    # Check date
+    date_check = dt.datetime.today().date()
+    today = date_check.strftime("%a %d %b, %Y")
+
+    form = OrderForm()
+    order_by = request.args.get("order_by")
+    if not order_by:
+        # Handle the case when the user does not select any option
+        # redirect the user to a default ordering
+        return redirect(url_for("get_all_rehearsals", order_by="desc"))
+    elif order_by not in ["asc", "desc", "created"]:
+        # Handle the case when the user is trying to manipulate the order_by parameter
+        # return an error message
+        flash("Invalid value for the order_by parameter")
+        return redirect(url_for("get_all_rehearsals", order_by="desc"))
+    else:
+
+        if order_by == "desc":
+            order_by_clause = Rehearsal.date.desc()
+        elif order_by == "created":
+            order_by_clause = Rehearsal.user_id.asc()
+        else:
+            order_by_clause = Rehearsal.date.asc()
+        rehearsals = Rehearsal.query.filter_by(user_id=current_user.id).order_by(order_by_clause)
```

Comment:
I was wondering if you could (and would want to) write it this way instead. Since SQLAlchemy lets you chain things together, it can be cool to build a query step by step.

```suggestion
        rehearsals = Rehearsal.query.filter_by(user_id=current_user.id)
        if order_by == "desc":
            rehearsals = rehearsals.order_by(Rehearsal.date.desc())
        elif order_by == "created":
            rehearsals = rehearsals.order_by(Rehearsal.user_id.asc())
        else:
            rehearsals = rehearsals.order_by(Rehearsal.date.asc())
```

Reply:
That is a great idea! It is still readable:)